### PR TITLE
Feat: @avalabs/avalanchejs upgrade and migration

### PR DESCRIPTION
**Pull request by Philippoes**
## Changed

- Migrate to @avalabs/avalanchejs
- adjust signing and address retrieval functionality for Avalanche
- Refactor, temporarily disable Ledger

### CODE REVIEW

File: src/avalanche.ts [hunk 1]
```diff
@@ -103,35 +42,29 @@ const getMnemonicWallet = async (
...
 
-    signC: async (tx: UnsignedEvmTx) => tx.sign(cchain.keyChain()),
-    signP: async (tx: UnsignedPlatformTx) => tx.sign(pchain.keyChain()),
+  const sha256Hash = createHash('sha256').update(publicKeyBuffer).digest();
+  const ripemd160Hash = createHash('ripemd160').update(sha256Hash).digest();
 
+  return {
+    ethereumAddress: await wallet.getAddress(),
+    signC: async (tx: EVMUnsignedTx) => sign(tx, wallet.privateKey),
+    signP: async (tx: UnsignedTx) => sign(tx, wallet.privateKey),
```
Comment (jdomingos) [inline src/avalanche.ts:56]:
```suggestion
    sign: async (tx: EVMUnsignedTx | UnsignedTx) => sign(tx, wallet.privateKey),
```

**Review by jdomingos — APPROVED** [review]
So first of all, splendid job!

In theory seems like we can have just on sign, because the code is the same. You just kept it separate because of the different types that feed it, right? But I guess we can abstract on the `src/avalanche.ts` just doing the suggestion I had. 

Other than that, can't fault it 👏👏👏👏

File: src/avalanche.ts (hunk 1 above)
Comment (Philippoes) [reply src/avalanche.ts]: Great suggestion

**Review by jdomingos — DISMISSED** [review]
🙇

**Pull request merged**